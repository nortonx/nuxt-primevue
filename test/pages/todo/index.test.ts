import { shallowMount } from "@vue/test-utils"
import Todo from "@/pages/todo/index.vue"
import { describe, it, expect, beforeEach } from 'vitest'
import { createTestingPinia } from '@pinia/testing'

describe('Todo page', () => {
  let wrapper: any
  beforeEach(() => {
    wrapper = shallowMount(Todo, {
      global: {
        plugins: [createTestingPinia()]
      }
    })
  })
  it('should render correctly and match snapshot', () => {

    expect(wrapper.html()).toMatchSnapshot()
  })
})