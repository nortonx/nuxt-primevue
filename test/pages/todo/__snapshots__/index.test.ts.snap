// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Todo page > should render correctly and match snapshot 1`] = `
"<div data-v-7167769e="" class="p-4 md:px-6 lg:px-8">
  <h1 data-v-7167769e="" class="mb-4 text-center">TO-DO List</h1>
  <div data-v-7167769e="" class="grid">
    <div data-v-7167769e="" class="col-8 col-offset-2">
      <div data-v-7167769e="" class="shadow-2 p-3 h-full flex flex-column surface-card">
        <input-text-stub data-v-7167769e="" modelvalue="" placeholder="What needs to be done?" class="w-full" variant="filled"></input-text-stub>
        <button-stub data-v-7167769e="" label="Add" class="w-full mt-3"></button-stub>
      </div>
    </div>
    <div data-v-7167769e="" class="col-8 col-offset-2">
      <div data-v-7167769e="" class="shadow-2 p-3 h-full flex flex-column surface-card">
        <div data-v-7167769e="" class="flex flex-column mx-4">
          <div data-v-7167769e="" class="text-900 font-bold text-xl text-center"> No items in the list </div>
        </div>
      </div>
    </div>
    <div data-v-7167769e="" class="col-8 col-offset-2">
      <div data-v-7167769e="" class="shadow-2 p-3 h-full flex flex-column surface-card">
        <div data-v-7167769e="" class="flex justify-content-between mx-4">
          <div data-v-7167769e="" class="text-900 font-bold text-xl"> Total: 0</div>
          <div data-v-7167769e="" class="text-900 font-bold text-xl"> Completed: 0</div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
